catroid/src/createatschool/java/org/catrobat/catroid/createatschool/ui/adapter/TemplateAdapter.java

```diff
@@ -110,14 +122,31 @@ public void onClick(View view) {
 	}
 
 	private void setImage(TemplateData templateData, ViewHolder holder) {
-		Picasso.with(context).load(templateData.templateResourceId).into(holder.image);
+		Picasso.with(context).load(getBaseUrl() + templateData.getThumbnail()).into(holder.image);
+	}
+
+	public String getBaseUrl() {
+		return !Strings.isNullOrEmpty(templateContainer.getBaseUrl())
+				? templateContainer.getBaseUrl() : Constants.MAIN_URL_HTTPS + "/";
+	}
+
+	@Override
+	public void onFetchTemplatesComplete(String templatesList) {
+		Gson gson = new Gson();
+		try {
+			templateContainer = gson.fromJson(templatesList, TemplateContainer.class);
+		} catch (JsonSyntaxException exception) {
+			ToastUtil.showError(context, context.getString(R.string.error_fetching_templates));
+		}
+
+		initTemplates();
 	}
 
 	public interface OnTemplateEditListener {
 		void onTemplateEdit(TemplateData templateData);
 	}
 
-	public static class ViewHolder {
+	private static class ViewHolder {
```

Comment:
This class isn't "completely unncessary". The ViewHolder is actually an important design pattern for performance optimization in Android --> see https://developer.android.com/training/improving-layouts/smooth-scrolling.html